torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2107,6 +2107,11 @@ std::shared_ptr<NCCLComm> ProcessGroupNCCL::getNCCLComm(
   // Get the device index
   auto deviceIndex = device.index();
   gpuGuard.set_index(deviceIndex);
+
+#if defined(NCCL_HAS_COMM_SPLIT) && defined(NCCL_HAS_COMM_NONBLOCKING)
+  options_->config.splitShare = 1;
```

Comment:
LGTM overall, but can you move this line under line 2115 so that we have a single block of code related to comm_split?

Reply:
@shuqiangzhang  Modification completed.